allennlp/data/data_loaders/multitask_data_loader.py

```diff
@@ -126,13 +104,10 @@ def __init__(
         self,
         reader: MultiTaskDatasetReader,
         data_path: Dict[str, str],
-        batch_size: int,
+        scheduler: MultiTaskScheduler,
```

Comment:
The biggest change here is that now the `MultiTaskScheduler` is responsible for making batches, not the data loader. When the data loader makes batches, but the scheduler sorts the instances, we can't guarantee things like homogeneous batches, so I changed it.

Reply:
This also means that the scheduler is no longer optional. You _have_ to specify one. Otherwise we don't even know a batch size.